packages/deployments/contracts/contracts_forge/messaging/connectors/admin/AdminHubConnector.t.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.17;
+
+import {MockSpokeConnector} from "../../../utils/Mock.sol";
+import {SpokeConnector} from "../../../../contracts/messaging/connectors/SpokeConnector.sol";
+import {AdminHubConnector} from "../../../../contracts/messaging/connectors/admin/AdminHubConnector.sol";
+import {RootManager} from "../../../../contracts/messaging/RootManager.sol";
+import {WatcherManager} from "../../../../contracts/messaging/WatcherManager.sol";
+import {MerkleTreeManager} from "../../../../contracts/messaging/MerkleTreeManager.sol";
+import {Message} from "../../../../contracts/messaging/libraries/Message.sol";
+import {RateLimited} from "../../../../contracts/messaging/libraries/RateLimited.sol";
+
+import "../../../utils/ForgeHelper.sol";
+
+contract AdminHubConnectorTest is ForgeHelper {
+  event MessageSent(bytes data, bytes encodedData, address caller);
+
+  using stdStorage for StdStorage;
+
+  // ============ Storage ============
+  uint32 BNB_DOMAIN = 6450786;
+  uint32 ETH_DOMAIN = 6648936;
+  address watcher = 0x43DB577bB3DD02989Dc3DC8e65E61a27d6914386;
+  address owner = 0x4d50a469fc788a3c0CdC8Fd67868877dCb246625;
+  SpokeConnector ETH_SPOKE_CONNECTOR = SpokeConnector(payable(0xF7c4d7dcEc2c09A15f2Db5831d6d25eAEf0a296c));
+
+  // ============ config
+  AdminHubConnector adminHubConnector;
+  RootManager rootManager = RootManager(0xd5d61E9dfb6680Cba8353988Ba0337802811C2e1);
+
+  // ============ Setup ============
+  function setUp() public {
+    uint256 forkId = vm.createSelectFork(
+      "https://eth-mainnet.g.alchemy.com/v2/CAtB_bTRwPneOAh_xLABggG3yUOR2wsV",
+      // 17528353
+      17464115
+    );
+    adminHubConnector = new AdminHubConnector(ETH_DOMAIN, BNB_DOMAIN, address(rootManager));
+  }
+
+  // ============ utils ============
+
+  function test_AdminHubConnector__connectAndProve_works() public {
```

Comment:
this needs to be an integration level test, not just on forge.

even if the message is processed twice, the `InboxFacet` prevents the funds from being minted twice. however, the `proveAndProcess` function would succeed, and im concerned about a drain on relayer fees if old messages are reattempted.

because this is a system-level test, we need to test on testnet. but because it is majorly a concern if *spoke connectors* are replaced, this testing is likely not incredibly impactful

Reply:
but we should have unit tests